### PR TITLE
[Packaging] Bump embedded Python version to 3.11.5

### LINKED ISSUE #26740: Bundled Python version has multiple vulnerabilities
### Describe the bug

Release 2.49 installed on Windows with Python.exe version 3.10.10 which has the below 3 vulnerabilities.

CVE-2023-0464 https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-0464
CVE-2023-0466 https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-0466
CVE-2023-24329 https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-24329

image

### Related command

az --version

Listed Python version has multiple vulnerabilities

### Errors

No error in output. SDK update required. 

### Issue script & Debug output

No Issue in script . SDK update required. 

### Expected behavior

The SDK should include the latest python version 3.10.12 which addresses these issues. 

https://www.python.org/downloads/release/python-31012/

### Environment Summary

azure-cli                         2.49.0

core                              2.49.0
telemetry                          1.0.8

Dependencies:
msal                              1.20.0
azure-mgmt-resource               22.0.0

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\gbarnes1\.azure\cliextensions'

Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:05:00) [MSC v.1929 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bebound**
**Description**

Fix #26740, CVE-2023-0464, CVE-2023-0466, CVE-2023-24329 are fixed in Python 3.11.4

**The user may need to reinstall their extensions if the dependency provides different wheels for different python version.**

PS: `scripts/release/debian/Dockerfile` is only used in `scripts/release/debian/README.md`, which is also deprecated. So I don't update its version number.

Related issue: https://github.com/Azure/azure-cli/issues/24494

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

**Comment by bebound** [comment]
Close as Python 3.10.12 does not contains binary installers.
Python 3.10 is now in the "security fixes only" stage of its life cycle, which pushes us to use Python 3.11 instead.

https://www.python.org/downloads/release/python-31012/

File: scripts/release/homebrew/docker/formula_generate.py [hunk 1]
```diff
@@ -124,6 +125,10 @@ def update_formula() -> str:
     resp.raise_for_status()
     text = resp.text
 
+    # update python version
+    text = re.sub('depends_on "python@.*"', f'depends_on "python@{PYTHON_VERSION}"', text, 1)
+    text = re.sub(r'virtualenv_create\(libexec, "python.*"', f'virtualenv_create(libexec, "python{PYTHON_VERSION}"', text, 1)  # pylint: disable=line-too-long
+
```
Comment (jiasli) [inline scripts/release/homebrew/docker/formula_generate.py:131]: Should these lines be dropped in the next release? It is really inconvenient to make any change to the formula.

Or, can we just keep the formula as is and make the change when a PR is submitted for https://github.com/Homebrew/homebrew-core/blob/master/Formula/a/azure-cli.rb ?

In my thought, the Azure CLI repo should be self-contained - it should never depend on external resources to build and install.

Comment (bebound) [reply scripts/release/homebrew/docker/formula_generate.py:131]: I prefer to keep it here for two reasons.
1. `TestHomebrewFormula` can check whether new formula works.
2. Homebrew users can use the new Python version as soon as possible.

For other changes like the `without_pip: false` in https://github.com/Azure/azure-cli/pull/27186, we can create a PR for homebrew.

Sometimes the Homebrew community update our formula, which cause a conflict, so we modify the external formula. This behavior is introduced in https://github.com/Azure/azure-cli/pull/11964

**Review by jiasli — APPROVED** [review]

**Pull request merged**